src/main/java/com/echobox/api/linkedin/jsonmapper/LinkedInJsonMapper.java

```diff
@@ -0,0 +1,155 @@
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.echobox.api.linkedin.jsonmapper;
+
+import static java.util.Collections.unmodifiableList;
+
+import com.echobox.api.linkedin.logging.LinkedInLogger;
+
+import org.apache.commons.lang3.StringUtils;
+import org.json.JSONArray;
+import org.json.JSONException;
+import org.json.JSONObject;
+import org.slf4j.Logger;
+
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ * LinkedIn implementation of a JSON-to-Java mapper.
+ * 
+ * @author Joanna
+ *
+ */
+public class LinkedInJsonMapper extends DefaultJsonMapper {
+
+  private static Logger LOGGER = LinkedInLogger.getLoggerInstance();
+
+  /**
+   * Creates a JSON mapper which will throw
+   * {@link com.echobox.api.linkedin.exception.LinkedInJsonMappingException} whenever an error
+   * occurs when mapping JSON data to Java objects.
+   */
+  public LinkedInJsonMapper() {
+    super();
+  }
+
+  /**
+   * Creates a JSON mapper which delegates to the provided {@code jsonMappingErrorHandler} for
+   * handling mapping errors.
+   * 
+   * @param jsonMappingErrorHandler
+   *          The JSON mapping error handler to use.
+   * @throws IllegalArgumentException
+   *           If {@code jsonMappingErrorHandler} is {@code null}.
+   * @since 1.6.2
+   */
+  public LinkedInJsonMapper(JsonMappingErrorHandler jsonMappingErrorHandler) {
+    super(jsonMappingErrorHandler);
+  }
+
+  /**
+   * @see com.echobox.api.linkedin.JsonMapper#toJavaList(java.lang.String, java.lang.Class)
+   */
+  @Override
+  public <T> List<T> toJavaList(String json, Class<T> type) {
+    if (type == null) {
+      throw new LinkedInJsonMappingException("You must specify the Java type to map to.");
+    }
+
+    json = StringUtils.trimToEmpty(json);
+
+    if (StringUtils.isBlank(json)) {
+      if (jsonMappingErrorHandler.handleMappingError(json, type, null)) {
+        return null;
+      }
+      throw new LinkedInJsonMappingException("JSON is an empty string - can't map it.");
+    }
+
+    if (json.startsWith("{")) {
+      // LinkedIn may return an empty object as a list - handle it here
+      if (isEmptyObject(json)) {
+        if (LOGGER.isTraceEnabled()) {
+          LOGGER
+              .trace("Encountered {} when we should've seen []. Mapping the {} as an empty list and"
+                  + "moving on...");
+        }
+        return new ArrayList<>();
+      }
+
+      // LinkedIn returns arrays wrapped in an object
+      // Extract the values out of the JSON response
+      try {
+        JSONObject jsonObject = new JSONObject(json);
+        int total = jsonObject.optInt("_total");
+        JSONArray values = jsonObject.optJSONArray("values");
+        if (values == null) {
+          if (LOGGER.isTraceEnabled()) {
+            LOGGER.trace("No values provided in the JSON, carrying on...");
```

Comment:
perhaps this should be a error if the total is > 0?

Reply:
Not sure if we can rely on that value - what if LinkedIn returns total = 0, but has values? Then we should use those values (probably won't happen but can do to make sure) - happy to discuss

Reply:
OK happy to keep the values as the source of truth, we will still get an error is the `total != values.length` so we will know when it is happening.

Reply:
Just logged so it's ok